hw/lib/signup/sign_up_screen.dart

```diff
@@ -0,0 +1,313 @@
+              child: Container (
+                alignment: Alignment.center,
+                child: SingleChildScrollView(
+
+                  child: Column (
+                      children: [
+                        Row(
+                          children: [
+                            Assets.button.image(width: 89, height: 88),
+                            Padding(
+                              padding: const EdgeInsets.only(left: 55.0),
+                              child: Text(Batch16String.current.createAccount,
+                                style: TextStyle(fontSize:  18, fontWeight: FontWeight.w500),textAlign: TextAlign.center,),
+                            ),
+                          ],
+                        ),
+                        Padding(
+                          padding: const EdgeInsets.all(8.0),
+                          child: Assets.female.image(width: 89, height: 88),
+                        ),
+                        Text( Batch16String.current.createAccount, style: TextStyle(fontWeight: FontWeight.w400, fontSize: 12),),
+                        Row(
+                          mainAxisAlignment: MainAxisAlignment.center,
+                          children: [
+
+                            BlocListener<SignUpCubit, SignUpState>(
+                              listener: (context, state) {
+                                if (state is SignUpFailed  && state.provider =='github') {
+                                  ScaffoldMessenger.of(context).showSnackBar(SnackBar(content: Text(Batch16String.current.notSupported('Github'))));
+                                }
+                              },
+                              child: AccountHolder(image: Assets.git.image(width:  43, height:  43),
+                                  onTap: () {
+                                    context.read <SignUpCubit>().signupWithGitHub();
+
+
+                                  } ),
+                            ),
+                            AccountHolder(image: Assets.google.image(width:  43, height:  43),
+                              onTap: () =>
+                                  context.read<SignUpCubit>().signupWithGoogle()
+
+                              ,),
+                            AccountHolder(image: Assets.linkedIN.image(width:  43, height:  43),
+                                onTap: () =>
+                                    context.read<SignUpCubit>().signupWithLinkedIn())
+
+                          ],
+                        ),
+                        BlocSelector<SignUpCubit, SignUpState, String?>(
+                          selector: (state) {
+                            if (state is SignUpFailed)   {
+                              return state.provider;
+                            }
+                            return null;
+
+                          },
+                          builder: (context, state) {
+                            if (state == 'google') {
```

Comment:
Requirement: clicking on Google icon will add a text display below the 3 icons row (above text ‘or’): Google sign up is not supported

this can be done with 1 of following options:

1. use specific state for google provider

```
BlocBuilder<SignUpCubit, SignUpState>(
  builder: (context, state) {
    if (state is GoogleSignupFailed) {
        return Text(Batch16String.current.notSupported(state.provider);
    }
    return SizedBox.shrink();
  }
)
```

and in state, define new state
```
class GoogleSignupFailed extends SignUpState {
  final String provider;

  GoogleSignupFailed(this.provider);

  List<Object?> props => [provider];
}
```

and in cubit, emit this state
```
  void signupWithGoogle() {
    emit(GoogleSignupFailed(provider: 'google'));
  }

```
2. use the same state (as you're using now), and use BlocBuilder to build it

```
BlocBuilder<SignUpCubit, SignUpState>(
  builder: (context, state) {
    if (state is SignUpFailed && state.provider == 'google') {
        return Text(Batch16String.current.notSupported(state.provider);
    }

    return SizedBox.shrink();
  }
)
```